src/create_agent/check_agent.jl

```diff
@@ -0,0 +1,50 @@
+"""
+
+Function for checking the structure of the agent
+"""
+function check_agent(agent::Agent)
+
+    if length(agent.shared_parameters) > 0
+
+        ## Check for the same derived parameter in multiple shared parameters 
+        #Get out the derived parameters of all shared parameters 
+        derived_parameters = [
+            parameter for list_of_derived_parameters in [
+                agent.shared_parameters[i].derived_parameters for
+                i in keys(agent.shared_parameters)
+            ] for parameter in list_of_derived_parameters
+        ]
+
+        #check for duplicate names
+        if length(derived_parameters) > length(unique(derived_parameters))
+            #get a tuple of the occurances of the parameters and their names to take out the repeated parameters
+            repeated_parameters = [
```

Comment:
Hmm jeg tror det er værd at undgå bøvlet kode hvis vi kan.
Man skulle tro der var en nemmere måde at finde de ting i en vector der er flere af.
Hvis der ikke er, syns jeg måske vi bare skal sige
if length(derived_parameters) > length(unique(derived_parameters))
throw(ArgumentError("at least one parameter is being set by multiple shred parameters. This is not supported."))

Reply:
yes, det lyder fint. Jeg var bare på killing spree med list comprehensions